.gitattributes

```
@@ -1,2 +1,2 @@
 *.java text=auto
-*.xml text=auto
+*.xml text=lf
```

```suggestion
*.xml text=auto
```

If you're on Windows you can use the following Git settings instead:

```bash
git config --global core.autocrlf false
git config --global core.eol crlf
```

---

I think that happend when i tried to rebase my branch to the head of openhab-addons but i dont know how.
I try to figure out how to remove it without just reverting the rebase.

---

I really dont knwo what happend, it think it just got somehow messed up in the github history. If i do git rebase -i c9d5c07ef583bf567e91b53d588406756f572e77 (SHA1 of my last commit), the ecobee commit does not show up, only my commits. Also, if you look at the commit history of my last commit after the rebase, it is not showing there, only in the one before. So i think its OK ?

---

The changes are still part of your PR, see:

https://github.com/openhab/openhab-addons/pull/14186/files
https://github.com/openhab/openhab-addons/pull/14186/commits

---

OK thanks i see it now. I tried to remove it using "git rebase -i", but every way i tried gave me dozens of merge conflicts.
I am afraid i am making it worse if i keep fiddling around without really understanding what i am doing. The only safe way i can think of is to delete everything including my fork and the pull request, copy/paste my binding over into a new one and create a new pull request.  But that would loose the commit history. The other way would be to just leave it as it is since the ecobee commit that somehow ended up in my commits is part of PR  #14170 that already got merged, so its in the main anyway and should not cause any harm. 
If you have any ideas help is appreciated! :-)

---

Maybe this comment can help? https://github.com/openhab/openhab-addons/pull/13570#issuecomment-1372691793

---

I wouldn't bother too much about the branch history because it will be squashed later on anyways. :wink:

Recreating the "hapero" branch on your computer, based on the most recent openhab `main` branch and copying back your changes should also normally work. Then you can "force push" your local "hapero" branch back to your fork and you won't have recreate the PR. It will just replace all the commits that show up on here on GitHub.

But if you are not a Git ninja do whatever works for you as long as you make sure all your code does not get lost. :slightly_smiling_face:

---

Thank you for the help!
I followed your suggestion and created a new local branch, Copy/pasted my changes into it, comitted and force pushed to the origin. That seems to have cleaned things up.
Thanks again, i did not know that you could overwrite the commit history like that.

---

Great! It seems to have worked out well and it builds in a couple of minutes now too. 👍

---

That was suprising. At first i was afraid i broke something again, like accidentally removing all other bindings, because the build was so fast now :-)